### PR TITLE
Generate per-element topology/configs.

### PR DESCRIPTION
**Pull request by kormat**
The generated directory structure looks like this:

```
gen/
    ISD1/
        AD10/
            bs1-10-1/       # Configuration for bs1-10-1 element
                ad.conf
                path_policy.conf
                supervisord.conf
                topology.conf
                certs/      # Certs for AD1-10
                    ISD1-V0.crt
                    ISD1-AD10-V0.crt
                keys/
                    ad-sig.key  # Signing key for AD1-10
            common/    # Common ad.conf and topology.conf
```

Each element has its own fully independant config directory, with copies
of the AD's configs/certs/keys, and a supervisord.conf that lists only
the element. This allows easy deployment of an element on a machine.

There's a common/ subdir at the AD level, for use by sciond and anything
else that needs access to AD-wide config.

Also:
- Make all servers only require a server id, and an optional config dir
  argument, on the command line.
- Unify main() for all servers.
- Improved cert chain request/reply logging
- Simplify Topology.get_own_config, and improve error handling.
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/439%23issuecomment-151196865%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23issuecomment-151207172%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23discussion_r43094363%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23discussion_r43094664%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23issuecomment-151417475%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23discussion_r43098693%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23discussion_r43098698%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23issuecomment-151196865%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Still%20needs%20unit%20tests%20for%20%60lib.main%60%2C%20working%20on%20them%20now.%22%2C%20%22created_at%22%3A%20%222015-10-26T16%3A27%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%60lib.main%60%20unit%20tests%20added.%22%2C%20%22created_at%22%3A%20%222015-10-26T16%3A53%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22otherwise%20lgtm%21%22%2C%20%22created_at%22%3A%20%222015-10-27T08%3A54%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20fca7936aa2c66f69e55c08cafa4a499c20555355%20infrastructure/beacon_server.py%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23discussion_r43094363%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22sort%20imports%22%2C%20%22created_at%22%3A%20%222015-10-27T08%3A39%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222015-10-27T09%3A31%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/beacon_server.py%3AL48-55%22%7D%2C%20%22Pull%20fca7936aa2c66f69e55c08cafa4a499c20555355%20infrastructure/cert_server.py%2023%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/439%23discussion_r43094664%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22ditto.%22%2C%20%22created_at%22%3A%20%222015-10-27T08%3A44%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222015-10-27T09%3A31%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/cert_server.py%3AL17-44%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/439#issuecomment-151196865'>General Comment</a></b>
- <a href='https://github.com/kormat'></a> Still needs unit tests for `lib.main`, working on them now.
- <a href='https://github.com/kormat'></a> `lib.main` unit tests added.
- <a href='https://github.com/pszalach'></a> otherwise lgtm!
- [ ] <a href='#crh-comment-Pull fca7936aa2c66f69e55c08cafa4a499c20555355 infrastructure/beacon_server.py 20'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/439#discussion_r43094363'>File: infrastructure/beacon_server.py:L48-55</a></b>
- <a href='https://github.com/pszalach'></a> sort imports
- <a href='https://github.com/kormat'></a> Done
- [ ] <a href='#crh-comment-Pull fca7936aa2c66f69e55c08cafa4a499c20555355 infrastructure/cert_server.py 23'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/439#discussion_r43094664'>File: infrastructure/cert_server.py:L17-44</a></b>
- <a href='https://github.com/pszalach'></a> ditto.
- <a href='https://github.com/kormat'></a> Done

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/439?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/439?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/439'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: infrastructure/beacon_server.py [hunk 1]
```diff
@@ -51,7 +48,7 @@
     SCIONParseError,
     SCIONServiceLookupError,
 )
-from lib.log import init_logging, log_exception
+from lib.main import main_wrapper, main_default
```
Comment (pszal) [inline infrastructure/beacon_server.py]: sort imports

File: infrastructure/cert_server.py [hunk 2]
```diff
@@ -17,31 +17,27 @@
...
 from lib.crypto.certificate import TRC
 from lib.defines import CERTIFICATE_SERVICE, SCION_UDP_PORT
-from lib.log import init_logging, log_exception
 from lib.packet.cert_mgmt import (
     CertChainReply,
     CertChainRequest,
     TRCReply,
     TRCRequest,
 )
+from lib.main import main_wrapper, main_default
```
Comment (pszal) [inline infrastructure/cert_server.py]: ditto.

File: infrastructure/beacon_server.py (hunk 1 above)
Comment (kormat) [inline infrastructure/beacon_server.py]: Done

File: infrastructure/cert_server.py (hunk 2 above)
Comment (kormat) [inline infrastructure/cert_server.py]: Done

**Pull request merged**